### ISSUE TITLE
Possible Bug in ast-util.arr

### ISSUE THREAD

**Issue by belph** [brownplt/pyret-lang#660]
Looking at [the source](https://github.com/brownplt/pyret-lang/blob/a5d99d0e9a1ace2e6f3504d62e2b913dd97949eb/src/arr/compiler/ast-util.arr#L72) for `merge-nested-blocks` in `ast-util.arr`, I notice that the `for` loop visits each statement `s` when passing it to the `cases` statement, but it appends the *unvisited* `s` to the accumulator. This seems wrong, since any nested blocks contained within non-blocks will still be un-merged in the result.

**Comment by blerner** [comment]
That does seem wrong.  Can you trigger buggy behavior with an actual source program?  If so, _definitely_ add it as a regression test, and fix it!

**Comment by blerner** [comment]
(somehow this comment didn't get commented, last week...)

We've just looked into this a bit more, and instrumented ast-util.arr to print if it ever encounters the interesting line of merge-nested-blocks.  So far it hasn't.

**Comment by schanzer** [comment]
@blerner does that mean it's dead code?

**Comment by blerner** [comment]
Dunno.  Probably, but I don't recall this in detail.

**Comment by schanzer** [comment]
If the instrumentation has been there for two years, and hasn’t reported anything...

**Comment by shriram** [comment]
That assumes the instrumentation output is where someone can see it _and_ is looking at it…

**Comment by blerner** [comment]
I don't think we merged instrumentation into the mainline compiler, since we didn't want to break anything in use. We just had tested it locally, and fairly lightly.
